include/localization.js

```diff
@@ -397,15 +401,18 @@ module.exports = function LocalizationModule(pb) {
         if (!Localization.keys[key]) {
             return finalize(options.defaultVal);
         }
-        else if (this.cache[key]) {
+        else if (self.cache[key]) {
 
             //we have already processed this key once for this instance
-            return finalize(processValue(this.cache[key]));
+            return finalize(processValue(self.cache[key]));
         }
 
         //key create key path
         var keyBlock = Localization.storage;
         var parts = key.split(Localization.KEY_SEP);
+        if(keyBlock[self.site] && keyBlock[self.site][parts[0]])
+            keyBlock = keyBlock[self.site];
```

Comment:
the above lines are what make localizations site aware. I know Brian was going to refactor the way this works so any thoughts on a better method of doing this would be appreciated.